test/functional/feature_governance.py

```diff
@@ -52,131 +91,148 @@ def run_test(self):
+        assert_equal(self.nodes[0].getblockchaininfo()["softforks"]["v20"]["bip9"]["status"], "started")
+        self.check_superblockbudget(False)
+
         proposal_time = self.mocktime
-        sb_block_height = self.nodes[0].getblockcount() + 10 - self.nodes[0].getblockcount() % 10
-        p0_payout_address = self.nodes[0].getnewaddress()
-        p1_payout_address = self.nodes[0].getnewaddress()
-        p2_payout_address = self.nodes[0].getnewaddress()
-        p0_amount = float(1.1)
-        p1_amount = float(3.3)
-        p2_amount = float(self.nodes[0].getsuperblockbudget(sb_block_height)) - p1_amount
-
-        p0_collateral_prepare = self.prepare_object(1, uint256_to_string(0), proposal_time, 1, "Proposal_0", p0_amount, p0_payout_address)
-        p1_collateral_prepare = self.prepare_object(1, uint256_to_string(0), proposal_time, 1, "Proposal_1", p1_amount, p1_payout_address)
-        p2_collateral_prepare = self.prepare_object(1, uint256_to_string(0), proposal_time, 1, "Proposal_2", p2_amount, p2_payout_address)
+        self.p0_payout_address = self.nodes[0].getnewaddress()
+        self.p1_payout_address = self.nodes[0].getnewaddress()
+        self.p2_payout_address = self.nodes[0].getnewaddress()
+        self.p0_amount = satoshi_round("1.1")
+        self.p1_amount = satoshi_round("3.3")
+        self.p2_amount = self.expected_v20_budget - self.p1_amount
+
+        p0_collateral_prepare = self.prepare_object(1, uint256_to_string(0), proposal_time, 1, "Proposal_0", self.p0_amount, self.p0_payout_address)
+        p1_collateral_prepare = self.prepare_object(1, uint256_to_string(0), proposal_time, 1, "Proposal_1", self.p1_amount, self.p1_payout_address)
+        p2_collateral_prepare = self.prepare_object(1, uint256_to_string(0), proposal_time, 1, "Proposal_2", self.p2_amount, self.p2_payout_address)
 
         self.nodes[0].generate(6)
+        self.bump_mocktime(6)
         self.sync_blocks()
 
         assert_equal(len(self.nodes[0].gobject("list-prepared")), 3)
         assert_equal(len(self.nodes[0].gobject("list")), 0)
 
-        p0_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p0_collateral_prepare["hex"], p0_collateral_prepare["collateralHash"])
-        p1_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p1_collateral_prepare["hex"], p1_collateral_prepare["collateralHash"])
-        p2_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p2_collateral_prepare["hex"], p2_collateral_prepare["collateralHash"])
+        self.p0_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p0_collateral_prepare["hex"], p0_collateral_prepare["collateralHash"])
+        self.p1_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p1_collateral_prepare["hex"], p1_collateral_prepare["collateralHash"])
+        self.p2_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p2_collateral_prepare["hex"], p2_collateral_prepare["collateralHash"])
```

Comment:
nit: would have been nice if these "self." only changes were in their own commit for easier review